### ISSUE TITLE
Mejorar el engagement en twitter

### ISSUE THREAD

**Issue by nukeador** [PucelaBits/websegura#52]
**¿Está relacionado con un problema? Por favor detallalo.**
Creo que sería importante para mejorar el engagement que los visitantes y que más gente diera a compartir en twitter, ya que es la única forma de visualizar el problema. Actualmente menos del 5% lo hace.

![imagen](https://user-images.githubusercontent.com/531906/106821521-99127180-667d-11eb-9145-dd55a3dda7b1.png)

**Describe la solución que te gustaría ver**

-  En la página de detalle de una web, podríamos tener un contador de cuantos tuits con el usuario +#websegura se han publicado. De esta forma animaría a más gente a unirse o si es cero, a empezar la campaña en redes.
- En las tablas generales un CTA más destacado para que la gente sepa que la meta es compartir en twitter, ¿quizá mostrando un minicontador numérico al lado de cada botón con cuantos tuits hay?

Desconozco si [el API de twitter](https://developer.twitter.com/en/docs/twitter-api/tweets/search/introduction) permite mucha magia, pero bajar un json con los tuits nos permitiría decir algo como:

> Tuitear (7)

>  25 personas están exigiendo una mejora
>  [avatar] [avatar] [avatar] [avatar] [avatar] [avatar] 
>  ¡ÚNETE!

**Describe las alternativas posible**
Mostrar todos los tuits con el usuario +#websegura en la parte lateral del detalle de una web.

**Comment by nukeador** [comment]
Otras plataformas que hacen algo parecido con las "firmas"

![imagen](https://user-images.githubusercontent.com/531906/106614348-860f7c80-656b-11eb-908e-887e21bd0abf.png)

**Comment by nukeador** [comment]
Otra idea: mostrar el % de webs inseguras de una provicia y mostrar 2 o 3 que queremos que la gente se enfoque en tuitear

> Valladolid tiene un 1% de webs seguras, ¡ayudanos a subir este porcentaje tuiteando a los siguientes organismos!
> [barrita de progreso]
> Sitio1 - Tuitear
> Sitio2 - Tuitear
> Sitio3 - Tuitear

**Comment by nukeador** [comment]
Otro ejemplo/idea para llamar la atención de la gente.

![imagen](https://user-images.githubusercontent.com/531906/107078771-19a4af80-67ef-11eb-8001-6817fd6804d3.png)

Al dar ayudar te mostraría los enlaces para tuitear directamente a los 3-4 que queramos hacer foco.

**Comment by nukeador** [comment]
Una primera aproximación se ha implementado

![image](https://user-images.githubusercontent.com/531906/107234274-c7989f80-6a23-11eb-95a5-8ad783aa8e8d.png)

**Comment by fcoterroba** [comment]
> 
> 
> **¿Está relacionado con un problema? Por favor detallalo.**
> Creo que sería importante para mejorar el engagement que los visitantes y que más gente diera a compartir en twitter, ya que es la única forma de visualizar el problema. Actualmente menos del 5% lo hace.
> 
> ![imagen](https://user-images.githubusercontent.com/531906/106821521-99127180-667d-11eb-9145-dd55a3dda7b1.png)
> 
> **Describe la solución que te gustaría ver**
> 
>     * En la página de detalle de una web, podríamos tener un contador de cuantos tuits con el usuario +#websegura se han publicado. De esta forma animaría a más gente a unirse o si es cero, a empezar la campaña en redes.
> 
>     * En las tablas generales un CTA más destacado para que la gente sepa que la meta es compartir en twitter, ¿quizá mostrando un minicontador numérico al lado de cada botón con cuantos tuits hay?
> 
> 
> Desconozco si [el API de twitter](https://developer.twitter.com/en/docs/twitter-api/tweets/search/introduction) permite mucha magia, pero bajar un json con los tuits nos permitiría decir algo como:
> 
> > Tuitear (7)
> 
> > 25 personas están exigiendo una mejora
> > [avatar] [avatar] [avatar] [avatar] [avatar] [avatar]
> > ¡ÚNETE!
> 
> **Describe las alternativas posible**
> Mostrar todos los tuits con el usuario +#websegura en la parte lateral del detalle de una web.

Creo que se puede hacer algo chulo en Python, y como dicen por ahí, "Si puedes imaginarlo, puedes programarlo" Indistintamente del lenguaje, jajajaja
En resumen, asignámelo, si quieres y vemos qué hacer!

**Comment by nukeador** [comment]
Genial @fcoterroba aunque supongo que al final tendríamos que pasarlo a js como el resto de la web para no añadir más dependencias. @adrm @palmerabollo ¿mejor no?

La idea es consultar el API y guardarlo en un json local en _data/ que luego podemos usar desde el framework.

**Comment by adrm** [comment]
Sí, sería ideal mantener todo con un solo lenguaje, simplifica bastante el proyecto. El problema que veo aquí es la gestión de las claves de la API de twitter si lo tenemos todo público, pero seguro que hay alguna solución para mantener las claves privadas en otro repo o algo así y que se puedan usar desde las github actions de este repo.

**Comment by nukeador** [comment]
se puede tener repo privado que solo tire las consultas a twitter y desde este hacer checkout solo de los datos?

**Comment by palmerabollo** [comment]
Las github actions permiten configurar ["secretos"](https://docs.github.com/en/actions/reference/encrypted-secrets) (p.e. claves de la api de twitter) que no son públicas, sólo son visibles a quienes administran el repositorio y se referencian en los workflows como `${{ secret.LoQueSea }}`. Ese no sería un gran problema.

**Comment by palmerabollo** [comment]
Tiene pinta de ser bastante complicado, pero estaría interesante sacar las respuestas oficiales de "estamos arreglándolo", etc (ejemplos https://twitter.com/EMTmadrid/status/1360144592562176003 y https://twitter.com/AuvasaVLL/status/1360128570870358016)

**Comment by nukeador** [comment]
@fcoterroba Se que estabas con esto. ¿Necesitas ayuda de algún tipo? ¿Hasta dónde podemos sacar del API?

**Comment by palmerabollo** [comment]
Por si hace falta ayuda, este sería el punto de entrada al API, que permite buscar por hashtag @websegura https://developer.twitter.com/en/docs/twitter-api/tweets/search/quick-start/recent-search.

Cualquiera de estas libs recubren el API y se integrarían bien con lo que tenemos ya montado en nodejs con eleventy y el crawler:
- https://github.com/draftbit/twitter-lite
- https://github.com/HunterLarco/twitter-v2

Para usarlas hace falta crear una app de twitter y usar el consumer_key y consumer_secret que dan (la parte más tediosa, el api v2 creo aún está en "beta", nos vale el v1).

**Comment by adrm** [comment]
Reabro que me da la impresión de que esto se cerró automáticamente por error.

**Comment by palmerabollo** [comment]
Sí, fue por error, gracias por reabrirla. Te he citado en #186 @adrm , está a punto de caramelo, sólo quedaría enganchar en eleventy.js que lea el _data/twitter/summary.json y lo meta en el _data/all.json con el resto de info.

**Comment by nukeador** [comment]
Alguna idea más sobre cómo mostrar las metas de tuits, en un bloque para la página de detalle (donde tenemos ahora lo de tuitear)

![imagen](https://user-images.githubusercontent.com/531906/110308428-618e4080-8000-11eb-92dc-3f1e9dd8c9b8.png)

El texto podría decir algo como "Consigamos la siguiente meta de tuits" o algo así

**Comment by adrm** [comment]
El change.org de la seguridad web, me encanta jajaja

**Comment by palmerabollo** [comment]
Hola. Yo veo bien probar el bloque en la pantalla de detalle (no sé si tenemos datos de cuánta gente llega a esa pantalla), y probar si tiene efecto. Podría haber varias categorías (1er tweet, 1-5 tweets, 6-10 ...) con nombres, para ir incentivando ser el primero, llegar a 5, llegar a 10... Esta semana no voy a tener mucho tiempo, y de desarrollo web me quedé un poco en los 90, a ver si alguien se anima.
